sat/cli/bmccreds/parser.py

```diff
@@ -102,6 +128,26 @@ def add_bmccreds_subparser(subparsers):
              '--password.'
     )
 
+    bmccreds_subparser.add_argument(
+        '--password-length',
+        help='The desired length of the random BMC password. Only valid with '
+             '--random-password.'
+    )
+
+    bmccreds_subparser.add_argument(
+        '--password-chars',
+        help=f'A comma-separated list of allowed character sets to be used in '
+             f'a randomly generated password. Only valid with --random-password. '
+             f'Character sets are as follows: "alpha" includes all uppercase and '
+             f'lowercase ASCII letters, "numeric" includes digits 0-9, and "symbols" '
+             f'includes the symbols "{VALID_CHAR_SETS["symbols"].replace("%", "%%")}". '
+             f'If unspecified, defaults to all character sets mentioned above.'
+             f'You can specify explict characters also in case one of the preset options '
+             f'(alpha, numeric, or symbols) is insufficient. '
+             f'For example: sat bmccreds --password-chars "abcdefghijklmnopqrstuvwxyz!$%_".',
```

Comment:
I would maybe consider having two different options. One option that lets them choose from predefined character sets like "alpha", "numeric", and "symbols" as described in the help above. Then a different option that lets them specify an explicit set of characters. I can see two benefits to this approach:

1. It would make it harder for a typo in the name of a character set to go unnoticed. For example, what if the user types `--password-chars alpa,numerric`? In that case, the current code would just assume they mean the character set `alpanumeric` not the set of all alphabetical characters and numeric characters. That would result in an insecure password being generated.
2. It would make it possible to specify the comma character as an allowed character in an explicit character set. This is more of a minor concern, but it's a nice added benefit because the other option wouldn't treat the comma as a special delimiting character.

Possible option names could be `--password-char-sets` (basically the option I had implemented here initially) and `--password-chars` (the explicit lists of characters that you added here).

Reply:
Addressed